app/models/user/User.scala

```diff
@@ -141,22 +130,25 @@ case class User(
               if (userTrust != Default) {
                 Future.successful(userTrust)
               } else {
+                val projectsTable = TableQuery[ProjectTableMain]
+                val orgaTable     = TableQuery[OrganizationTable]
+                val memberTable   = TableQuery[OrganizationMembersTable]
+                val roleTable     = TableQuery[OrganizationRoleTable]
 
-                val projectMembersTable = TableQuery[ProjectMembersTable]
-                val orgaTable           = TableQuery[OrganizationTable]
-
-                val memberTable = TableQuery[OrganizationMembersTable]
-                val roleTable   = TableQuery[OrganizationRoleTable]
-
-                // TODO review permission logic
                 val query = for {
-                  pm <- projectMembersTable if pm.projectId === pScope.projectId // Join members of project
-                  o  <- orgaTable if pm.userId == o.userId // Filter out non organizations
-                  m  <- memberTable if m.organizationId === o.id
-                  r  <- roleTable if m.userId === r.userId && r.organizationId === o.id
+                  p <- projectsTable if pScope.projectId.bind === p.id
+                  o <- orgaTable if p.userId === o.id
+                  m <- memberTable if m.organizationId === o.id && m.userId === this.userId.bind
+                  r <- roleTable if this.userId.bind === r.userId && r.organizationId === o.id
                 } yield r.roleType
 
-                service.doAction(query.to[Set].result).map(biggestRoleTpe)
+                service.doAction(query.to[Set].result).map { roleType =>
+                  if (roleType.contains(RoleType.OrganizationAdmin) || roleType.contains(RoleType.OrganizationOwner)) {
```

Comment:
Why are we not always calling `biggestRoleType` here? Shouldn't that method be the one that checks for Org roles?

Reply:
No. We only want to inherit permissions from orgs if it's an orga owner or an orga admin. `biggestRoleTpe` is a method on the `User` class which gets the role with the highest trust, it's more general and not only for use in this case.